pulldownlayout/src/main/java/com/w4lle/library/PullDownView.java

```diff
@@ -0,0 +1,244 @@
+
+    private boolean isShowTopView;
+
+    private boolean isonce;
+
+    private ComponentContainer.LayoutConfig layoutParams;
+
+    private int oldTopMargin;
+
+    private boolean isTouchOne;
+
+    private boolean isTop;
+
+    private OnPullChangeListerner onPullChangeListerner;
+
+    private boolean isChangeSpeed;
+
+    private float pullDownHeight;
+
+    private float pullUpHeight;
+
+    /**
+     * PullDownView Constructor.
+     *
+     *  @param context - context for PullDownView constructor
+     *
+     */
+    public PullDownView(Context context) {
+        this(context, null);
+        setLayoutRefreshedListener(this);
+        addDrawTask(this);
+        setTouchEventListener(this);
+        setScrolledListener(this);
+    }
+
+    /**
+     * PullDownView Constructor.
+     *
+     *  @param context - context for PullDownView constructor
+     *  @param attrs - attributes
+     *
+     */
+    public PullDownView(Context context, AttrSet attrs) {
+        this(context, attrs, "");
+        setLayoutRefreshedListener(this);
+        addDrawTask(this);
+        setTouchEventListener(this);
+        setScrolledListener(this);
+    }
+
+    /**
+     * PullDownView Constructor.
+     *
+     *  @param context - context for RainyView constructor
+     *  @param attrs - attributes
+     *  @param defStyleAttr - defStyle attribute
+     *
+     */
+    public PullDownView(Context context, AttrSet attrs, String defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init(context, attrs);
+        setLayoutRefreshedListener(this);
+        addDrawTask(this);
+        setTouchEventListener(this);
+        setScrolledListener(this);
+    }
+
+    private void init(Context context, AttrSet attrs) {
+        DisplayAttributes metrics = DisplayManager.getInstance().getDefaultDisplay(context).get().getAttributes();
+        screenHeight = metrics.height;
+        topViewHeight = screenHeight;
+        touchSlop = 22;
+        getAttrs(attrs);
+    }
+
+    private void getAttrs(AttrSet attrs) {
+        isChangeSpeed = AttrUtils.getBooleanFromAttr(attrs, "is_change_speed", false);
+        pullDownHeight = AttrUtils.getDimensionFromAttr(attrs, "pull_down_height", DEFAULTPULLDOWNHEIGHT);
+        pullUpHeight = AttrUtils.getDimensionFromAttr(attrs, "pull_up_height", DEFAULTPULLUPHEIGHT);
+    }
+
+    @Override
+    public void onRefreshed(Component component) {
+        if (!isonce) {
+            ComponentContainer parentView = (ComponentContainer) this.getComponentAt(0);
+            topView = parentView.getComponentAt(0);
+            containerView = parentView.getComponentAt(1);
+            topView.getLayoutConfig().height = topViewHeight;
+            isonce = true;
+            layoutParams = containerView.getLayoutConfig();
+            oldTopMargin = layoutParams.getMarginTop();
+        }
+    }
+
+    @Override
+    public void onDraw(Component component, Canvas canvas) {
+        // onDraw method
```

Comment:
If there is nothing to be implemented in the onDraw(), then implemented interface can be removed along with the overriden method.

Reply:
Done